src/requests/consumption.js

```diff
@@ -0,0 +1,1059 @@
+
+			taskMap = _.mapValues(taskMap, (externalId) => {
+				// Find the corresponding object from projectsTasksDetailsAfterInsert
+				const task = _.find(projectsTasksDetailsAfterInsert, { externalId: externalId })
+
+				// If found, replace externalId with _id; otherwise, keep the externalId
+				return task ? ObjectId(task._id) : externalId
+			})
+
+			templateProjects.forEach((project) => {
+				let projectTasks = []
+				project.tasks.forEach((task) => {
+					projectTasks.push(taskMap[task])
+				})
+				project.tasks = projectTasks
+			})
+
+			await projectsCollection.insertMany(templateProjects)
+		}
+
+		const projectTemplatesAfterInsert = await projectsCollection
+			.find({
+				externalId: {
+					$in: templateProjects.map((projects) => projects.externalId),
+				},
+			})
+			.toArray()
+
+		// Add a new 'type' key to each project
+		const updatedProjectTemplates = projectTemplatesAfterInsert.map((project) => ({
+			...project, // Spread the existing project fields
+			type: common.PROJECT,
+			resource_id: templateProjectsIdMap[project.externalId],
+		}))
+
+		return [...updatedProjectTemplates]
+	}
+}
+
+/**
+ * Format Program Template
+ * @name formatProgramTemplate
+ * @param {Object} templateData - Program template data
+ * @returns {Object} - Response contains formatted template
+ */
+const formatProgramTemplate = (templateData) => {
+	try {
+		let language = templateData?.language
+			? templateData?.language
+			: templateData?.resource
+			? templateData?.resource.flatMap((resource) => {
+					return resource.languages.map((language) => {
+						return language.label
+					})
+			  })
+			: []
+		language = [...new Set(language)]
+		let keywords = templateData?.keywords
+			? templateData?.keywords
+			: templateData?.resource
+			? templateData?.resource.flatMap((resource) => {
+					return resource.keywords.split(',')
+			  })
+			: []
+		keywords = [...new Set(keywords)]
+
+		const resourceDetails = templateData?.resource
+			.map((resource) => {
+				return {
+					_id: ObjectId(resource.published_id),
+					type: resource.type,
+				}
+			})
+			.filter((published_id) => published_id !== null && published_id !== undefined)
+		let scope = {
+			roles: [],
+			entityType: [],
+		}
+		let metaInformation = {
+			recommendedFor: [],
+		}
+		if (templateData?.targeting_criteria) {
+			templateData?.targeting_criteria.forEach((targeting) => {
+				const targeting_entity = targeting?.entity_targeting?.value
+
+				if (!scope.entityType.includes(targeting_entity)) scope.entityType.push(targeting_entity)
+				if (targeting?.roles) {
+					targeting.roles.forEach((role) => {
+						if (!scope.roles.includes(role.value)) scope.roles.push(role.value)
+						if (!metaInformation.recommendedFor.includes(role.label))
+							metaInformation.recommendedFor.push(role.label)
+					})
+				} else {
+					scope.roles = []
+					metaInformation.recommendedFor = []
+				}
+				targeting[targeting_entity].forEach((target) => {
```

Comment:
@adithyadinesh0412  we can optimize something like below
targets.forEach(target => {
            if (!scope[targetingEntity].includes(target._id)) {
                scope[targetingEntity].push(target._id);
            }
            if (!metaInformation[targetingEntity].includes(target.name)) {
                metaInformation[targetingEntity].push(target.name);
            }
        });